src/transformers/models/olmo/modeling_olmo.py

```diff
@@ -238,6 +270,16 @@ def __init__(self, config: OlmoConfig, layer_idx: Optional[int] = None):
         self.k_proj = nn.Linear(self.hidden_size, self.num_key_value_heads * self.head_dim, bias=config.attention_bias)
         self.v_proj = nn.Linear(self.hidden_size, self.num_key_value_heads * self.head_dim, bias=config.attention_bias)
         self.o_proj = nn.Linear(self.hidden_size, self.hidden_size, bias=config.attention_bias)
+        self.q_norm = (
+            get_layer_norm(config.layer_norm_type, self.num_heads * self.head_dim, config.rms_norm_eps)
+            if config.use_q_norm
+            else None
+        )
+        self.k_norm = (
+            get_layer_norm(config.layer_norm_type, self.num_key_value_heads * self.head_dim, config.rms_norm_eps)
+            if config.use_k_norm
+            else None
+        )
```

Comment:
hey! This goes a bit against the `transformers` philosophy: we never change an old model to support a new architecture! We need a `Olmo2` model 🤗 
with modular this should be fairly simple to implement!

Reply:
Ok, I've re-implemented the model as a new model using modular. I'll put out the PR once we have decided on a suitable model name internally (we don't intend to call this release OLMo 2).